### ISSUE TITLE
'no-undef' is not thrown for undeclared/unassigned object members

### ISSUE THREAD

**Issue by invicticide** [eslint/eslint#10461]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.9.0
* **npm Version:** 6.1.0

Running ESLint via the [ESLint VS Code extension](https://github.com/Microsoft/vscode-eslint) and also with [eslint-loader](https://github.com/webpack-contrib/eslint-loader#readme) for webpack.

**Please show your full configuration:**

```js
{
    "extends": "eslint:recommended",
    "parserOptions": {
        "sourceType": "module"
    },
    "env": {
        "browser": true,
        "node": true
    }
}
```

**Source code sample**

```js
bar(); // eslint throws no-undef here, as expected

var foo = {};
foo.bar(); // eslint thinks this is perfectly valid (???)
```

**What did you expect to happen?**

The call to `foo.bar()` should've thrown `no-undef`... right?

It's unclear to me if this is actually a bug, or if there's just some other rule I need to enable (I looked through the rules but other than `no-undef` I didn't see anything that seemed relevant...?)

Or maybe this is just something that's outside the scope of any linter...? After all, in a more complex example there could've been other code that defined `foo.bar()` in a different place and potentially in a weird way (e.g. `foo["bar"] = function() { ... }` or whatever) so maybe my expectations are just out of whack?

**Comment by invicticide** [comment]
Cool, thanks. I thought this might be the case but I wanted to be sure, and I was having trouble finding this question discussed anywhere else.
